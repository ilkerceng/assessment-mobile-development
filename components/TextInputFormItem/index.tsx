import React, { useState, useEffect } from "react";
import { StyleSheet, Text } from "react-native";
import InputLabel from "../input-label";
import TextInput from "../text-input";
import { validate } from "../../utils/validations";

const TextInputFormItem = (props) => {
  const {
    formData,
    caption,
    validator,
    onValidation,
    ...textInputProps
  } = props;

  const [errors, setErrors] = useState({
    [caption]: {
      message: "",
    },
  });

  useEffect(() => {
    if (validator) {
      validateInput("");
    }
    return () => {};
  }, []);

  useEffect(() => {
    onValidation({ field: errors[caption].message });
    return () => {};
  }, [errors]);

  const validateInput = (value: string) => {
    let validateResult = {
      [caption]: {
        message: validate(value, validator),
      },
    };
    setErrors(validateResult);
    onValidation({ field: errors[caption].message });
  };

  const onChange = (value: string) => {
    formData.value = value;
    validateInput(value);
  };

  return (
    <>
      <InputLabel id={caption} label={caption} />
      <TextInput onChange={onChange} {...textInputProps} />
      {errors[caption].message ? (
        <Text style={styles.validationText}>{errors[caption].message}</Text>
      ) : null}
    </>
  );
};

const styles = StyleSheet.create({
  validationText: {
    color: "red",
  },
});

export default TextInputFormItem;
