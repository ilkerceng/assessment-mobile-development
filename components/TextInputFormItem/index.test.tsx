import React from "react";
import renderer from "react-test-renderer";
import TextInputFormItem from ".";

const selectProps = {
  type: "input",
  caption: "E-mail",
  validator: "email",
};

describe("<text-input-form-item />", () => {
  test("match snapshot", () => {
    const tree = renderer
      .create(<TextInputFormItem onChange={() => false} {...selectProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
