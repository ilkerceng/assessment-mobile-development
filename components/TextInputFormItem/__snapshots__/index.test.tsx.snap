// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<text-input-form-item /> match snapshot 1`] = `
Array [
  <View>
    <Text>
       
      E-mail
       
    </Text>
  </View>,
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      }
    }
    type="input"
    underlineColorAndroid="transparent"
    value=""
  />,
]
`;
